src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -334,10 +336,24 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
   }
 
   override def getUserProxy(userEmail: WorkbenchEmail): Future[Option[WorkbenchEmail]] = {
-    directoryDAO.loadSubjectFromEmail(userEmail).map {
+    directoryDAO.loadSubjectFromEmail(userEmail).flatMap {
       // don't attempt to handle groups or service accounts - just users
-      case Some(user: WorkbenchUserId) => Option(toProxyFromUser(user))
-      case _ => None
+      case Some(user: WorkbenchUserId) => getUserProxy(user)
+      case _ => Future.successful(None)
+    }
+  }
+
+  def getUserProxy(userId: WorkbenchUserId): Future[Option[WorkbenchEmail]] = {
+    directoryDAO.readUserAttribute[String](userId, Attr.proxyEmail).map(_.map(WorkbenchEmail))
+  }
+
+  private def withProxyEmail[T](userId: WorkbenchUserId)(f: WorkbenchEmail => Future[T]): Future[Option[Future[T]]] = {
```

Comment:
Turns out I think it's going to be better to treat a missing proxy group as an exceptional case to avoid having to deal with futures and options together. Change to this function is coming soon.